client/signup/steps/site-picker/site-picker-submit.jsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { PLAN_FREE } from 'lib/plans/constants';
+import { getSite } from 'state/sites/selectors';
+import SignupActions from 'lib/signup/actions';
+
+class SitePickerSubmit extends Component {
+	componentWillMount() {
+		const {
+				stepSectionName,
+				stepName,
+				goToStep,
+				selectedSite
+			} = this.props,
+			hasPaidPlan = selectedSite && selectedSite.plan && selectedSite.plan.product_slug !== PLAN_FREE;
+
+		SignupActions.submitSignupStep(
+			{
+				stepName,
+				stepSectionName,
+				siteId: selectedSite.ID,
+				siteSlug: selectedSite.slug
+			},
+			[],
+			{}
+		);
+
+		SignupActions.submitSignupStep( { stepName: 'themes', wasSkipped: true }, [], {
+			themeSlugWithRepo: 'pub/twentysixteen'
+		} );
+
+		if ( hasPaidPlan ) {
+			SignupActions.submitSignupStep( { stepName: 'plans', wasSkipped: true }, [], { cartItem: null, privacyItem: null } );
+
+			goToStep( 'user' );
+		} else {
+			goToStep( 'plans' );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default connect(
+	( state, ownProps ) => {
+		return {
+			selectedSite: getSite( state, ownProps.siteSlug )
```

Comment:
That's part of a broader conversation for which people take different sides. 😄 For context, `getSite` [received support](https://github.com/Automattic/wp-calypso/commit/ef40e66e73a2bde0e0e791ebdb50b8c6f7862205#diff-4aaad6137f1bc91a0f39e5b5e3ddb478R74) for slugs recently, during the First Great Killing of SitesList. An ongoing debate was what we expect of selectors, for instance:
- Should we prefer more "macro" entity-based selectors like `getSite` that already include most data one would want out of it as object properties (`site.slug`, `site.is_previewable`, etc.)? Or should we prefer smaller, dedicated, numerous selectors for many of these data (`isPreviewable( state, siteId )`)?
- Should we allow polymorphism in selectors for convenience? (e.g. `getSite` accepting IDs, slug strings, maybe domains, etc.)

You can always chat with @mtias if you're feeling philosophical about this. (Or about anything at all, really.)